### ISSUE TITLE
eslint binary not installed locally in 7.12.1 (works in 7.9.0)

### ISSUE THREAD

**Issue by Domiii** [eslint/eslint#13822]
**Tell us about your environment**

* **ESLint Version:** 7.12.1
* **Node Version:** 12.18.4
* **npm Version:** yarn 1.22.5
* **OS:** Mac OS 10.14.6

Also I use Lerna (not sure if it might interfere with things here).

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```bash
eslint -v
npx --no-install eslint -v
```

**What did you expect to happen?**

It should show me the version.

**What actually happened? Please include the actual, raw output from ESLint.**

I get:

> $ eslint -v
> eslint: command not found

> $ npx --no-install eslint -v
> not found: eslint

**Are you willing to submit a pull request to fix this bug?**

Uhm... Not sure?

**Analysis + Workaround**

It seems like the install is entirely incomplete. There is no `lib` folder, no `bin` folder, no nothing:

> $ ls node_modules/eslint/**
> node_modules/eslint/CHANGELOG.md        node_modules/eslint/package.json
> node_modules/eslint/README.md
> 
> node_modules/eslint/conf:
> category-list.json      replacements.json
> 
> node_modules/eslint/messages:
> all-files-ignored.txt                   plugin-conflict.txt
> extend-config-missing.txt               plugin-invalid.txt
> failed-to-read-json.txt                 plugin-missing.txt
> file-not-found.txt                      print-config-with-directory-path.txt
> no-config-found.txt                     whitespace-found.txt
> 
> node_modules/eslint/node_modules:
> ansi-styles             doctrine                semver
> chalk                   eslint-visitor-keys     shebang-command
> color-convert           globals                 shebang-regex
> cross-spawn             has-flag                supports-color
> debug                   path-key                which

**Workaround**

When reverting to `eslint@7.9.0` (I used that specific version because it works on another project of mine), everything is fine and dandy:

> $ ls ./node_modules/eslint/**
> ./node_modules/eslint/CHANGELOG.md      ./node_modules/eslint/README.md
> ./node_modules/eslint/LICENSE           ./node_modules/eslint/package.json
> 
> ./node_modules/eslint/bin:
> eslint.js
> 
> ./node_modules/eslint/conf:
> category-list.json      default-cli-options.js  eslint-recommended.js
> config-schema.js        eslint-all.js           replacements.json
> 
> ./node_modules/eslint/lib:
> api.js          eslint          options.js      shared
> cli-engine      init            rule-tester     source-code
> cli.js          linter          rules
> 
> ./node_modules/eslint/messages:
> all-files-ignored.txt                   plugin-conflict.txt
> extend-config-missing.txt               plugin-invalid.txt
> failed-to-read-json.txt                 plugin-missing.txt
> file-not-found.txt                      print-config-with-directory-path.txt
> no-config-found.txt                     whitespace-found.txt
> 
> ./node_modules/eslint/node_modules:
> ansi-styles     cross-spawn     globals         semver          supports-color
> chalk           debug           has-flag        shebang-command which
> color-convert   doctrine        path-key        shebang-regex

**Comment by nzakas** [comment]
This is most likely being caused by the way you're installing ESLint. How are you installing it? Are you using npm, Yarn, something else?

**Comment by Domiii** [comment]
@nzakas yarn, installed locally.
